test/address.test.js

```diff
@@ -198,16 +199,54 @@ define(function(require) {
           .query('a', 'a=b')
           .header('accept','application/json')
           .body({hello:'world!'})
+          .timeout(60)
           .on('done', cb)
 
         update()
 
         web.req.should.have.been.calledOnce
         web.req.should.have.been.calledWith(req)
-
         cb.should.have.been.calledOnce
       })
 
+      it('should return error when negative timeout value is provided', function() {
+        var cb = sinon.spy()
+          , responseBody = {
+              message: 'Invalid timeout: must be an integer value greater than 0'
+            }
+
+        response.statusCode = 400
+        response.body = responseBody
```

Comment:
I didn't quite understand what this was so I had to do a double take and make sure I understood how the tests actually function, and as it turns out this right here means we're not actually testing this functionality. This test, and the other one that tests for null timeout values, isn't actually testing to make sure the middleware blows up during the request – really all we're doing here is pretend making a request and returning an exact response.

This seems indicative of a larger problem with this test suite to be honest, where we're just testing the API but not actually the functionality. The reason I found this out is because I didn't expect there to be response, I figured that when the request is actually made an exception would be thrown. This is a jarring experience, but frankly, an invalid timeout value should be a hard error – not a response – such that it can be caught in development rather than production.

Here's what I recommend: move these tests that check the middleware into its own file, called middleware.test.js, and set it up like so:

```javascript
define(function(require) {
  var sinon = require('sinon')
    , nap = require('@websdk/nap')
    , address = require('address')
    , middleware = require('middleware')

  describe('Middleware', function() {
    it('should throw error when negative timeout value is provided', function() {
      var resource = sinon.spy()
      var cb = sinon.spy()
      var error

      try {
        address('/wibble/{id}')
          .web(
            nap.web()
              .use(middleware.requestTimeout)
              .resource('wibble', '/wibble/{id}', resource)
          )
          .param('id', '123')
          .timeout(-10)
          .on('done', cb)()
      } catch (e) {
        error = e
      } finally {
        cb.should.not.have.been.called
        resource.should.not.have.been.called
        error.message.should.equal('Invalid timeout: must be an integer value greater than 0')
      }
    })

    it('should return error when null timeout value is provided', function() {
      var resource = sinon.spy()
      var cb = sinon.spy()
      var error

      try {
        address('/wibble/{id}')
          .web(
            nap.web()
              .use(middleware.requestTimeout)
              .resource('wibble', '/wibble/{id}', resource)
          )
          .param('id', '123')
          .timeout(null)
          .on('done', cb)()
      } catch (e) {
        error = e
      } finally {
        cb.should.not.have.been.called
        resource.should.not.have.been.called
        error.message.should.equal('Invalid timeout: must be an integer value greater than 0')
      }
    })

    it('should set timeout to 30 seconds by default', function() {
      address('/wibble/{id}')
          .web(
            nap.web()
              .use(middleware.requestTimeout)
              .resource('wibble', '/wibble/{id}', assertDefaultTimeout)
          )
        .param('id', '123')
        .on('done', function() {})()

      function assertDefaultTimeout(req, res) {
        req.timeout.should.equal(30)
      }
    })
  })
})
```

This will test the middleware proper, since this is more of an integration test than an API test, like those in address.test.js. Of course, this also means removing the middleware tests from address.test.js.